src/librustdoc/html/render/context.rs

```diff
@@ -651,7 +651,22 @@ impl<'tcx> FormatRenderer<'tcx> for Context<'tcx> {
                      <script defer src=\"{root_path}settings{suffix}.js\"></script>",
                     root_path = page.static_root_path.unwrap_or(""),
                     suffix = page.resource_suffix,
-                )
+                );
+                // Pre-load all theme CSS files, so that switching feels seamless.
+                //
+                // When loading settings.html as a popover, the equivalent HTML is
+                // generated in main.js.
+                for file in &shared.style_files {
+                    if let Ok(theme) = file.basename() {
+                        write!(
+                            buf,
+                            "<link rel=\"preload\" href=\"{root_path}{theme}{suffix}.css\" \
```

Comment:
Instead of generating all this HTML and duplicating the logic, why not creating a JS function to load them and instead call this function here like:

```html
<script>window.preloadAllCss();</script>
```

? That will make the maintenance easier too to only have it in one place.

Reply:
And you would call this function in the `setTimeout` call too.

Reply:
Makes sense. We'll need to be careful if we ever update this but I think it's fine otherwise. Thanks for the extra explanation!